opensearch/src/test/java/org/opensearch/sql/opensearch/storage/OpenSearchIndexScanTest.java

```diff
@@ -160,10 +164,8 @@ public OpenSearchResponse answer(InvocationOnMock invocation) {
                   when(response.isEmpty()).thenReturn(false);
                   ExprValue[] searchHit = searchHitBatches[batchNum];
                   when(response.iterator()).thenReturn(Arrays.asList(searchHit).iterator());
-                } else if (batchNum == totalBatch) {
-                  when(response.isEmpty()).thenReturn(true);
                 } else {
-                  fail("Search request after empty response returned already");
+                  when(response.isEmpty()).thenReturn(true);
```

Comment:
Since we now fetch results in batches, additional search requests can be made when calling `hasNext()`